components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java

```diff
@@ -65,18 +79,19 @@ public Response apisPost(APIDTO body,String contentType){
 
         boolean isTenantFlowStarted = false;
         try {
-            org.wso2.carbon.apimgt.api.model.API apiToAdd = fromDTOtoAPI(body);
-             String tenantDomain =  CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-             if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
-                 isTenantFlowStarted = true;
-                 PrivilegedCarbonContext.startTenantFlow();
-                 PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
-             }
-
-             provider.addAPI(apiToAdd);
-
-             //how to add thumbnail
-             //publish to external stores
+            org.wso2.carbon.apimgt.api.model.API apiToAdd = APIMappingUtil.fromDTOtoAPI(body);
+            String loggedInUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            APIProvider apiProvider = RestApiUtil.getProvider(loggedInUser);
+            String tenantDomain =  CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
```

Comment:
The tenants flows have been moved currently from impl level to CXF handler level
Fixed comments in #1091 commit : 897ef52